src/index/vchordrq/types.rs

```diff
@@ -44,6 +53,8 @@ pub struct VchordrqInternalBuildOptions {
     #[serde(default = "VchordrqInternalBuildOptions::default_build_threads")]
     #[validate(range(min = 1, max = 255))]
     pub build_threads: u16,
+    #[serde(default = "VchordrqInternalBuildOptions::default_kmeans_algorithm")]
+    pub kmeans_algorithm: KMeansAlgorithm,
```

Comment:
Dimension reduction can be combined with any specific K-means algorithm. It is unreasonable to couple dimension reduction with performing K-means on the partitions.

Reply:
Now the reduction dimension can be used with both `lloyd` or `approx`, and is being seperated from algorithm. The syntax is like this, I think it could be refined later.
```sql
CREATE INDEX ON t USING vchordrq (val vector_l2_ops)
WITH (options = $$
residual_quantization = false
[build.internal]
lists = [3]
spherical_centroids = false
kmeans_dimension_reduction = 2
[build.internal.kmeans_algorithm.approx]
```